src/app/services/rssFolder.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from "@angular/core";
+import { AuthService } from "../auth.service";
+import { GraphService } from "../graph.service";
+
+@Injectable()
+export class RssFolderService {
+  RSSGEN_FOLDER_NAME = "rssCaster";
+  newFolderName = "";
+  selectedFolder = "rssCaster";
+  constructor(
+    private authService: AuthService,
+    private graphService: GraphService
+  ) {}
+
+  async rssgenRootFolderExists() {
+    const client = this.graphService.getClient();
+    return await client
+      .api("/me/drive/root/children")
+      .post({
+        name: this.RSSGEN_FOLDER_NAME,
+        folder: {
+          view: {
+            sortBy: "name",
+            sortOrder: "ascending",
+            viewType: "default"
+          }
+        },
+        "@microsoft.graph.conflictBehavior": "fail"
+      })
+      .then(() => true)
+      .catch(err => {
+        return err.code === "nameAlreadyExists";
+      });
+  }
+
+  async getCurrentFolders() {
+    const client = this.graphService.getClient();
+    const currentFolders = await client
+      .api(`/me/drive/root:/${this.RSSGEN_FOLDER_NAME}:/children`)
+      .get()
+      .then(response => {
+        let folderNames = response.value.map(folder => folder);
+        return folderNames;
+      });
```

Comment:
minor: is this supposed to map out `folder.name`?  If not it can combine into:
```suggestion
      .then(response => response.value);
```
Or with foldername 
```suggestion
      .then(response => response.value.map(folder => folder.name));
```